controller/lib/hal/hal.h

```diff
@@ -269,6 +295,35 @@ inline uint16_t HalApi::serialBytesAvailableForWrite() {
   return Serial.availableForWrite();
 }
 
+inline void HalApi::watchdog_init() {
+#ifdef ARDUINO_AVR_UNO
+  // FIXME Does this pose potential issues for arduino code updates?
+
+// The device will be reset if watchdog_handle is not called within roughly this
+// time period.
+//
+// Options are: 15ms, 30ms, 60ms, 120ms, 250ms, 500ms, 1s, 2s, 4s, 8s.
+//
+// TODO: This value was not chosen carefully.
+#define WATCHDOG_TIMEOUT
+  wdt_enable(WDTO_120MS);
+#endif
+}
+[[noreturn]] inline void HalApi::reset_device() {
```

Comment:
Looks like this does nothing on native. It might be okay for now, but what are your thoughts on whether we need to mock the watchdog more substantially on native, and if yes, how might we do it?

Reply:
Oh that's a good point; in fact, it's wrong to mark it as noreturn if it actually returns on native.

Maybe it should be a mock method on native, then from gtest we can check if it was called?  This is the (untested) HAL_MOCK macro at the top of the file.

Reply:
I think just mocking these methods is not enough - we would need to mock the passage of time too. And we would need to be clear about what we're actually trying to test regarding watchdog on native, or are we just getting the code to compile. Probably best left to a separate PR + issue, because it deserves some discussion.

Reply:
What do we test against in this case? That watchdog is being called in timely manner? It might be very complicated to do as you would have to imitate all aspects of running on real hardware.